github-upload-public-key/run.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+set -e
+
+CODER_ACCESS_URL="${CODER_ACCESS_URL}"
+CODER_OWNER_SESSION_TOKEN="${CODER_OWNER_SESSION_TOKEN}"
+
+if [ -z "$CODER_ACCESS_URL" ]; then
+  echo "No coder access url specified!"
+  exit 1
+fi
+
+if [ -z "$CODER_OWNER_SESSION_TOKEN" ]; then
+  echo "No coder owner session token specified!"
+  exit 1
+fi
+
+if [ -z "$GITHUB_TOKEN" ]; then
+  echo "No GITHUB_TOKEN in the workspace environment!"
+  echo "The user must be authenticated with Github before this script can be run."
+  exit 1
+fi
```

Comment:
Even better, you can use `coder external-auth access-token github` to automatically fetch the token for a provider! Then it'd refresh as well.

Reply:
Yup, `GITHUB_TOKEN` env var method is only because the `gh` cli does not support a more dynamic approach. I've been pushing everyone in the community to not use when they do not have to.

Reply:
Yeah, we should remove this from dogfood.

Reply:
https://github.com/coder/coder/pull/13106